### PR TITLE
Add alternative emotes channel

### PR DESCRIPTION
**Pull request by BasedUser**
## About the PR
In reference to #15907 (incompatible), this PR adds an alternative emotes channel. This allows people to use both `*` and `@` for using emotes.

**Media**

https://user-images.githubusercontent.com/21104932/235433343-4f81e6aa-6496-477f-aebe-612e58e16594.mp4

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl: router
- add: Added an alternative emotes prefix, \*. Note that it doesn't get treated any differently than the normal @ prefix.

### CODE REVIEW

**Comment by BasedUser** [comment]
Right now, the client errors due to this:
```c#
    public static readonly Dictionary<ChatSelectChannel, char> ChannelPrefixes =
        PrefixToChannel.ToDictionary(kv => kv.Value, kv => kv.Key);
```
and the fact we have multiple Emotes resolutions, should I just copy the mapping above and leave out `SharedChatSystem.EmotesAltPrefix`?

**Comment by AJCM-git** [comment]
> Right now, the client errors due to this:
> 
> ```cs
>     public static readonly Dictionary<ChatSelectChannel, char> ChannelPrefixes =
>         PrefixToChannel.ToDictionary(kv => kv.Value, kv => kv.Key);
> ```
> 
> and the fact we have multiple Emotes resolutions, should I just copy the mapping above and leave out `SharedChatSystem.EmotesAltPrefix`? Or should I make a reverse mapping in-place anyway, but with a different method?

I think maybe the best way would be to add proper support for alternative chars by making the prefixes arrays of chars

**Comment by BasedUser** [comment]
> I think maybe the best way would be to add proper support for alternative chars by making the prefixes arrays of chars

why not strings?

**Comment by AJCM-git** [comment]
> > I think maybe the best way would be to add proper support for alternative chars by making the prefixes arrays of chars
> 
> why not strings?

i think working with char arrays makes more explicit the purpose

**Comment by BasedUser** [comment]
i'm not coming back to this due to IRL reasons

**Comment by BasedUser** [comment]
upstream merged

File: Content.Shared/Chat/SharedChatSystem.cs [hunk 1]
```diff
@@ -15,6 +15,7 @@ public abstract class SharedChatSystem : EntitySystem
     public const char LOOCPrefix = '(';
     public const char OOCPrefix = '[';
     public const char EmotesPrefix = '@';
+    public const char EmotesAltPrefix = '*';
```
Comment (AJCM-git) [inline Content.Shared/Chat/SharedChatSystem.cs:18]: As said before, to future proof this, is there a big problem with making these char arrays?

**Review by AJCM-git — CHANGES_REQUESTED** [review]

File: Content.Shared/Chat/SharedChatSystem.cs (hunk 1 above)
Comment (BasedUser) [reply Content.Shared/Chat/SharedChatSystem.cs:18]: technically no but it needs a huge chunk of rewriting chat code, which is outside the scope of this PR (i'm not going to refactor chat for this)

```
    // the following code is problematic in SharedChatSystem
    public static readonly char[] RadioCommonPrefix = ";".ToCharArray();  // EncryptionKeySystem.cs:223 makes `key` nullable
    public static readonly char[] RadioChannelPrefix = ":".ToCharArray(); // same location, same nulling
```

**Review by AJCM-git — APPROVED** [review]

**Pull request merged**